src/CMakeLists.txt

```diff
@@ -99,6 +99,12 @@ function(valhalla_module)
     target_compile_options(${library} PUBLIC -ftest-coverage -fprofile-arcs)
   endif()
 
+  # On more modern x86 processors that support SSE2, specifying the compiler options -mfpmath=sse -msse2
+  # ensures all float and double operations are performed in SSE registers and correctly rounded.
+  # These options do not affect the ABI and should therefore be used whenever possible for
+  # predictable numerical results
+  target_add_compile_flags_if_supported(${library} PRIVATE -mfpmath=sse -msse2)
```

Comment:
It seems these options are required only for x86-32, and are enabled by default on x86-64 (see section on `-mfpmath=unit` in https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html). Maybe worthwhile mentioning it here. Better still, can we conditionally enable them only for x86-32? (if not, this is fine too)

Reply:
i agree with this, can we limit this intervention to 32bit x86 builds? it would be nice if we could specify this only for the 32bit build in CI tbh. does cmake allow us to generically add compiler flags with an of the `-D` options? i suppose we need to add our own option for that. that would be my preference. @PureW what do you think?

Reply:
it seems that these flags are not completel redundant with automatically generated ones on 64bit bit x86 platforms however, measuring the performance difference shows its always faster than defaults.